File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n=s.size();
        int res=0;
        for(int i=0;i<n;i++)
        {
            vector<bool>p(256,false);
            
                for(int j=i;j<n;j++)
                {
                    if(p[s[j]]==true)
                    {
                        break;
                    }
                    else
                    {
                        p[s[j]]=true;
                        res=max(res,j-i+1);
                        
                    }
                }
            p[s[i]]=false;
        }
        return res;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length();
        int res =0;
        Map<Character, Integer>charMapIndex = new HashMap<>();
        int start = 0;
        for(int end =0; end<n; end++) {
            char currChar = s.charAt(end);
            if(charMapIndex.containsKey(currChar)) {
                start = Math.max(start, charMapIndex.get(currChar) +1);
            }
            res = Math.max(res, end - start +1);
            charMapIndex.put(currChar, end);

        } 
        return res;       
    }
}